openedx/core/djangoapps/theming/templatetags/optional_include.py

```diff
@@ -0,0 +1,77 @@
+"""
+The functions in this module are based on the contents of
+https://github.com/django/django/blob/1.8.5/django/template/loader_tags.py --
+specifically, the do_include function. It has been modified as little as
+possible, in order to match the behavior of the {% include %} template tag,
+except for making it optional.
+"""
+# Because we want to match the original loader_tags.py file as closely as
+# possible, we should disable pylint so it doesn't complain about the violations
+# that are already in that file
+# pylint: skip-file
+from django.template.base import (
+    TemplateSyntaxError, Library, token_kwargs, TemplateDoesNotExist
+)
+from django.template.loader_tags import IncludeNode
+
+register = Library()
+
+
+class OptionalIncludeNode(IncludeNode):
+    def render(self, context):
+        try:
+            super(OptionalIncludeNode, self).render(context)
+        except TemplateDoesNotExist:
+            return ''
+
+
+@register.tag('optional_include')
```

Comment:
Sorry i missed the comment at the top.

The code for a wrapped node is simple, not sure about how to write the 50-line function the other way:
```
class OptionalIncludeNode(Node):
    def __init__(self, include_node, *args, **kwargs):
        super(OptionalIncludeNode, self).__init__(*args, **kwargs)
        self.include_node = include_node
    def render(self, context):
        try:
            return self.include_node.render(context)
        except TemplateDoesNotExist:
            return ''
```